src/colmap/scene/reconstruction.cc

```diff
@@ -43,6 +43,15 @@ namespace colmap {
 
 Reconstruction::Reconstruction() : max_point3D_id_(0) {}
 
+bool Reconstruction::Check() const {
+  for (const auto& image : images_) {
```

Comment:
I recommend moving this check to the test file, where its only use case is. No need to clutter the public interface of the reconstruction.

Reply:
Done.